lib/Promise/functions.php

```diff
@@ -28,7 +28,9 @@
  * If any of the given Promises fails, the returned promise will immediately
  * fail with the first Promise that fails, and its reason.
  *
- * @param Promise[] $promises
+ * @param Promise<int>[] $promises
+ *
+ * @return Promise<int>
```

Comment:
I need to understand what goes on here - phpstan insists that I write something for `TReturn` template from `Promise`
What is this supposed to describe/hint to the code analysis?

(there are a few places in this file where I needed to write similar to get phpstan to pass)

Reply:
I think at this point we can't know the types involved.

therefore I would declare 2 templates of type mixed and use one for the param and one for the return-type generic.

Reply:
I just changed `int` to `mixed` - we pass in an array of "mixed" `Promise` and get back a `Promise` that is "mixed". Is that an OK thing to do here? @staabm

Reply:
for this particular method thats fine. I think we cannot get better then that.